src/firebolt/async_db/cursor.py

```diff
@@ -61,6 +61,15 @@ class CursorState(Enum):
     CLOSED = 4
 
 
+class QueryStatus(Enum):
+    """Class for query responses on server-side async queries."""
+
+    RUNNING = 1
+    ENDED_SUCCESSFULLY = 2
+    ENDED_UNSUCCESSFULLY = 3
+    NOT_AVAILABLE = 4
```

Comment:
nit: Rename to NOT_READY. NOT_AVAILABLE might give users the impression that it's will never be available, but the case is it's just not ready yet.